pulp_ansible/app/tasks/collections.py

```diff
@@ -81,6 +82,27 @@ def sync(remote_pk, repository_pk, mirror):
     d_version = AnsibleDeclarativeVersion(first_stage, repository, mirror=mirror)
     d_version.create()
 
+    repo_version = repository.latest_version()
+    to_deprecate = []
+    to_deprecate_query = Q()
+    for collection, deprecated in first_stage.deprecations.items():
+        namespace, name = collection.split(".")
+        to_deprecate_query |= Q(namespace=namespace, name=name)
+
+    if to_deprecate_query:
+        collections_deprecate_true_qs = Collection.objects.filter(to_deprecate_query)
+        for collection in collections_deprecate_true_qs:
+            to_deprecate.append(
+                AnsibleCollectionDeprecated(repository_version=repo_version, collection=collection)
+            )
+
+        AnsibleCollectionDeprecated.objects.bulk_create(to_deprecate, ignore_conflicts=True)
+
+    AnsibleCollectionDeprecated.objects.filter(
+        repository_version=repo_version,
+        collection__pk__in=Collection.objects.exclude(to_deprecate_query),
```

Comment:
for readability I would move `Collection.objects.exclude(to_deprecate_query)` to a line earlier and be:

`non_deprecated_qs = Collection.objects.exclude(to_deprecate_query)`.